membership/interface.go

```diff
@@ -0,0 +1,19 @@
+package membership
+
+// Member defines a member of the membership. It can be used by applications to
+// apply specific business logic on Members. Examples are:
+// - Get the address of a member for RPC calls, both forwarding of internal
+//   calls that should target a Member
+// - Decissions to include a Member in a query via predicates.
+type Member interface {
```

Comment:
A separate `Membership` interface over `swim` its self gives benefits due to separation of concern. Historically the hashring has been very tightly coupled to the swim states eg. Alive, Suspect, Faulty ... and other assumptions like address and identity being the same.

This interface defines a separation of concern describing what a Membership looks like on a higher level. Only including members that are actively participating, no view over faulty, leave and tombstone nodes any more.

Due to the lack of the membership package and interface the first predicates have been implemented in the swim package. Making projects that want to use predicates relying on the swim package which have a lot of hard to understand subtleties in there for applications.

Lastly with this interface we can do more experimentation with different implementations of membership protocols. Currently we have only swim, but it might be beneficial to experiment with a zookeeper or etcd backed membership protocol for a more consistent view of the online members. This could be configured transparent to the service.